source/validate_id.cpp

```diff
@@ -2606,7 +2671,8 @@ spv_result_t CheckIdDefinitionDominateUse(const ValidationState_t& _) {
       const Instruction* variable = _.FindDef(phi->word(i));
       const BasicBlock* parent =
           phi->function()->GetBlock(phi->word(i + 1)).first;
-      if (variable->block() && !variable->block()->dominates(*parent)) {
+      if (variable->block() && parent->reachable() &&
```

Comment:
Is it valid for a parent block to be not reachable (I see that this is the case in some of tests)?

Also this makes the comment above not accurate (// Check all OpPhi parent blocks are dominated by the variable...).